src/lib.rs

```diff
@@ -153,44 +211,6 @@ impl Response {
     }
 }
 
-async fn connect(credentials: &AzureCredentials) -> anyhow::Result<Arc<dyn ObjectStore>> {
-    let (account, container, key, host) = credentials.to_string_tuple();
-    let max_retries = CONFIG.get().unwrap().max_retries;
-    let retry_timeout = std::time::Duration::from_secs(CONFIG.get().unwrap().retry_timeout_sec);
-    let mut azure = MicrosoftAzureBuilder::new()
-        .with_account(account)
-        .with_container_name(container)
-        .with_access_key(key)
-        .with_retry(object_store::RetryConfig {
-            max_retries: max_retries,
-            retry_timeout: retry_timeout,
-            ..Default::default()
-        })
-        .with_client_options(object_store::ClientOptions::new()
-            .with_timeout(std::time::Duration::from_secs(20))
-            .with_connect_timeout(std::time::Duration::from_secs(10))
```

Comment:
do we still set `timeout` and `connect_timeout` somewhere? if not, why are we dropping that?

Reply:
We do set it through the dynamic JSON config that comes in that `Config` struct

Reply:
Where exactly? (sorry for my unfamiliarity with reading Rust)

Are we just assuming/relying on these being in the key-value pairs passed to the `builder.with_config(...)`? How does validation of these inputs work? like if i pass `connection_timeout: 10` (rather than `connect_timeout` or `ConnectTimeout` or whatever it is supposed to be)?

Reply:
Yes, we delegate all validation to the underlying object_store library by using the `builder.with_config(...)`. It disallows any unknown keys and validates if the values can be parsed into a `std::time::Duration` returning a proper error on build.

Reply:
ah great! love that it does validation for us - thanks!